src/mon/FSCommands.cc

```diff
@@ -904,6 +904,16 @@ class AddDataPoolHandler : public FileSystemCommandHandler
       }
     }
 
+    for (auto& fs : fsmap.get_filesystems()) {
+      // not allowed if poolid is already in use by any fs
+      const std::vector<int64_t> &data_pools = fs->mds_map.get_data_pools();
```

Comment:
you could change this to:

    const auto& data_pools = fs->mds_map.get_data_pools();

Reply:
Done